### ISSUE TITLE
When not tracking unused code, Psalm does not detect dead bug-alike expressions and function calls.

### ISSUE THREAD

**Issue by Ocramius** [vimeo/psalm#9028]
Take following example @ https://psalm.dev/r/e06c01d985 

```php
<?php

/** @psalm-pure */
function doSomething(): int { return 123; }

doSomething();

1 + 1;
```

When disabling unused code checks, this code produces no errors:

```
Psalm output (using commit 6353143): 

No issues!
```

This seems to be a recent change in `vimeo/psalm`, and I noticed it because of Psalm now requiring to declare `findUnusedCode` in `psalm.xml`.

In practice, I would argue that both expressions in the code example above are not "unused code", but actual bugs, where a developer may have forgotten an assignment.

This is **very** prominent when dealing with immutable objects, like in https://psalm.dev/r/50d14190c2 :

```php
<?php

$date = new DateTimeImmutable();

$date->modify('+1 day');
```

As a developer, having `findUnusedCode="false"` hide this kind of issue is really problematic :D

**Comment by orklah** [comment]
Are you sure something changed on that? 

As far as I remember, there was no change in that feature itself, we just forced people to set a value there (the value was false before so those were not reported)

**Comment by Ocramius** [comment]
Ah, sorry, indeed this is not a recent change, but the fact that people will probably switch to `findUnusedCode="false"` in the immediate future is going to lead to this being an actual problem.

In a very clean `level="1"` project that I'm working on with @davidebicego, for example, we have a lot of unused code that was hidden until `findUnusedCode="true"` was used: that flag is not usable right away, but setting `findUnusedCode="false"` is worse, because it hides unused expressions like the ones I've mentioned above.

What I'm suggesting here is that `findUnusedCode="false"` should probably **not** hide unused pure expressions :)

**Comment by orklah** [comment]
I guess for this specific case, the only thing to do would be to remove the flag check here: https://github.com/vimeo/psalm/blob/master/src/Psalm/Internal/Analyzer/Statements/Expression/Call/FunctionCallAnalyzer.php#L1080

There's 14 checks with this flag so maybe more would be concerned, you could try to PR that. If @weirdan is fine with it, we could merge that

It would need to be documented as this is basically UnusedCode that will be raised even though the flag is set to false
